pkg/controller/volume/persistentvolume/pv_controller.go

```diff
@@ -939,31 +934,39 @@ func (ctrl *PersistentVolumeController) updateVolumePhaseWithEvent(ctx context.C
 
 // assignDefaultStorageClass updates the claim storage class if there is any, the claim is updated to the API server.
 // Ignores claims that already have a storage class.
-// TODO: if resync is ever changed to a larger period, we might need to change how we set the default class on existing unbound claims
-func (ctrl *PersistentVolumeController) assignDefaultStorageClass(ctx context.Context, claim *v1.PersistentVolumeClaim) (bool, error) {
+func (ctrl *PersistentVolumeController) assignDefaultStorageClass(ctx context.Context, claim *v1.PersistentVolumeClaim) (*v1.PersistentVolumeClaim, error) {
```

Comment:
This is basically a revert of https://github.com/kubernetes/kubernetes/pull/113137

Do we really want to *not* resync? This was changed on purpose so we're sure `IsDelayBindingMode` and `findBestMatchForClaim` is executed again after updating PVC with a storage class and no further action is taken in by `syncUnboundClaim`

Reply:
You're right. I missed the reason.